server/lib/scheduler/Scheduler.ts

```diff
@@ -0,0 +1,110 @@
+import { filter } from "asyncro";
+
+class Scheduler {
+  private fetchAllWorkspaces;
+  private rootParentCache;
+  private schedule;
+
+  public constructor({ fetchAllWorkspaces, rootParentCache, schedule }) {
+    this.fetchAllWorkspaces = fetchAllWorkspaces;
+    this.rootParentCache = rootParentCache;
+    this.schedule = schedule;
+  }
+
+  public async getIdOfCurrentWorkspace(userId) {
+    const assignment = this.schedule.getMostRecentAssignmentForUser(userId);
+    return assignment.getWorkspaceId();
+  }
+
+  public async findNextWorkspace(userId) {
```

Comment:
I suggest renaming this to something like ``getNextBestWorkspaceForAssignment`` or similar. Then I'd remove the userId part of this and line 24, and have them either in a different function or outside of this class.

Reply:
It needs `userId` when it does the assignment.

Reply:
`findNextWorkspace` is just a mutation. It doesn't return anything. So the name doesn't seem apt, starting with "find".

Reply:
I'll change it to `assignUserToNextWorkspace`

Reply:
actually maybe just `assignNextWorkspace`, either way it's now clear it's doing something